cmd/installer/cli/api.go

```diff
@@ -36,10 +39,26 @@ type apiOptions struct {
 	WebAssetsFS     fs.FS
 }
 
-func startAPI(ctx context.Context, cert tls.Certificate, opts apiOptions, cancel context.CancelFunc) error {
-	listener, err := net.Listen("tcp", fmt.Sprintf(":%d", opts.ManagerPort))
-	if err != nil {
-		return fmt.Errorf("unable to create listener: %w", err)
+func startAPI(ctx context.Context, cert *tls.Certificate, opts apiOptions, cancel context.CancelFunc) error {
+	var listener net.Listener
+	var err error
+
+	if opts.SocketPath != "" {
```

Comment:
hmm why do we need this permutation though? the more permutations the more we'll need testing. can't we just use tcp for headless as well?

Reply:
This way we don't have to require port 30000 be available. Also don't have to generate certs. There are tests. Let me know if I should remove this.